WIP DM-5581 Document v1

jonathansick opened the pull request:
See http://sqr-011.lsst.io/en/tickets-dm-5581/

jonathansick committed:
Outline document sections

jonathansick committed:
Table of platforms and audiences

jonathansick committed:
Work in progress draft

jonathansick committed:
Work in progress: described LTD philosophy

frossie committed:
Edits

frossie committed:
More edits

jonathansick committed:
Restore whitespace conventions

If, after the second round of edits, the hard wrap camp wins I'll just
stick with that. For now, soft wrap is miles easier.

jonathansick committed:
Add Frossie as an author

jonathansick committed:
Remove 'strategy' from title

This document is really to capture the set of tools we use and how to
work together. Strategy oversells the scope of the document slightly.

ktlim commented on index.rst:
```diff
@@ -2,6 +2,613 @@
+tools so that information does not fragment and to boost the network
+effect. We have found that the parameter space of task, synchronicty
+and audience do require multiple tools, but we favour only having one
+supported tool for each locus in that space.
```
Would be nice to have a chart showing combinations of task, synchronicity, and audience and which tool satisfies each combination (possibly an extension of the chart below).

jonathansick replied:
Good idea. I'll make slides first, and see if such a chart can emerge from that.

ktlim commented on index.rst:
```diff
@@ -2,6 +2,613 @@
+continuous integration harness, is "out of sight out of mind" for the
+developers, and is hard to maintain. We are in the process of
+migrating all software documentation from Confluence to other, better
+harnesses.
```
These conditions don't seem to necessarily apply to more design-level documentation, where it's not obvious that the GitHub platform is better than tools that provide inline comment threads like Confluence and Google Docs.  (P.S.  Google Docs seems to be used a lot informally; should it be mentioned here somewhere?)

jonathansick replied:
What is Google Docs being used for?

ktlim replied:
I've seen it being used for 1) planning spreadsheets, 2) design documents (both within DM and LSST-wide in Operations planning and systems engineering).
Some people are also using Dropbox to pass around information like presentations.

timj replied:
I am assuming that Dropbox for presentations and google docs for design documents are not where the final thing ends up though? I think this discussion is where the results of the writing become permanent.

ktlim replied:
They're still communications mechanisms in use.

I don't think that it's yet clear that documents are generally migrating out of Google Docs.  I think I recall some that have turned into DocuShare Word docs.

timj replied:
I'd be pretty upset if google docs became the final thing that is linked to. Exporting to docushare is fine.

jbecla replied:
I think google docs have a place in our project, and they fill an important niche. We badly need place where we could keep things like presentations and all these little drawings and diagrams, and docushare is just not good for it. Yes we are putting the official docs in docushare (baseline docs, fdr presentations etc) but when it comes to less official docs, like all these qserv presentations, all these little diagrams that we used for baseline, they never made it there. One big issue with docushare is that is impossible to find things there, you have to retrieve to make any change etc. It is also so super convenient to share a doc, update it live, from mac, windows and linux. And that is just not there in docushare.

jonathansick committed:
Fill out overview of platforms and ethos

- communication as a shared responsibility
- Discuss information discovery; points of entry

jonathansick committed:
Zenodo, Docushare and ADS

jonathansick committed:
Description of Technotes, Controlled docs, Index

timj commented:
I have a comment in my inbox that @jbecla made on this PR on March 22 but it seems to have disappeared from the PR history. I really don't think that things should end up on Google Doc as the end location. They aren't discoverable there at all. How do we index them? What are these "Qserv presentations" that are being mentioned? We should make presentations more visible. One of the problems we have is that people treat presentations as transient things that just slowly disappear. I don't see how Google Docs is more discoverable than docushare. I absolutely agree that Google Docs is great for live sharing and updating in a cross platform manner. We need to solve the end game though. I'm putting all DM presentations that are presented externally on Zenodo. Maybe we need to talk about the persisting of internal presentations.

jonathansick commented:
Do we have any access-control concerns on our internal presentations? If not, then we can also put them on Zenodo. The goal of [DocHub](http://sqr-011.lsst.io/en/latest/#a-documentation-index) is to provide a unified LSST/DM API and front-end to everything in Zenodo, and elsewhere.

btw, should we have this conversation on https://community.lsst.org/t/technote-sqr-011-dm-communication-platforms/616?u=jsick since this is a closed PR?

ktlim commented:
@timj Jacek's comment shows up for me under "ktlim and 3 others commented on an outdated diff".
